transport/src/main/java/io/camunda/zeebe/transport/stream/impl/messages/StreamResponseDecoder.java

```diff
@@ -11,12 +11,18 @@
 import org.agrona.DirectBuffer;
 import org.agrona.concurrent.UnsafeBuffer;
 
-public final class StreamResponseReader {
+public final class StreamResponseDecoder {
   private final MessageHeaderDecoder headerDecoder = new MessageHeaderDecoder();
   private final DirectBuffer buffer = new UnsafeBuffer();
 
-  public <T extends StreamResponse> Either<ErrorResponse, T> read(
+  public <T extends StreamResponse> Either<ErrorResponse, T> decode(
       final byte[] bytes, final T response) {
+    // for backwards compatibility, accept empty responses as success
+    // to be removed with 8.5
```

Comment:
💭 Do we have any issue ofr something for this? Because I think it will just stay forever like this.

Reply:
> What do you mean? With 8.5, we could remove this behavior since customers shouldn't skip minor versions.

Sure but who does this or who keeps track of it? I don't think that a comment in code is enough. We have several things lying around which never gets removed. I would like to avoid this.

Reply:
I can create a follow up to do once code is frozen.

Reply:
https://github.com/camunda/zeebe/issues/15386